cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -239,6 +239,21 @@ def launch_kubernetes_dashboard():
 @when_not('kube-dns.available')
 def start_kube_dns(sdn_plugin):
     ''' State guard to starting DNS '''
+
+    # Interrogate the cluster to find out if we have at least one worker
+    # that is capable of running the workload.
+
+    cmd = ['kubectl', 'get', 'nodes']
+    try:
+        out = check_output(cmd)
+        if not out:
```

Comment:
This could literally return "invalid" and would still pass. Could you provide what you expect from good output and do something like
```python
if out or 'string' in out:
    # profit
else:
    # return
```